src/main/java/com/github/curriculeon/AdvancedPhoneBook.java

```diff
@@ -1,45 +1,71 @@
 package com.github.curriculeon;
 
-import java.util.List;
-import java.util.Map;
+import java.util.*;
+import java.util.stream.IntStream;
 
 /**
  * Created by leon on 1/23/18.
  */
 public class AdvancedPhoneBook {
+    private Map<String, List<String>> phoneBook = new LinkedHashMap<>();
+    private List<String> listOfPhoneNumbers = new ArrayList<>();
     public AdvancedPhoneBook(Map<String, List<String>> map) {
+        phoneBook = map;
     }
 
     public AdvancedPhoneBook() {
-        this(null);
+        //this(null);
     }
 
     public void add(String name, String phoneNumber) {
+        listOfPhoneNumbers.add(phoneNumber);
+        phoneBook.put(name, listOfPhoneNumbers);
     }
 
     public void addAll(String name, String... phoneNumbers) {
+        for (String s: phoneNumbers)
+            listOfPhoneNumbers.add(s);
+        phoneBook.put(name, listOfPhoneNumbers);
     }
 
     public void remove(String name) {
+        phoneBook.remove(name);
     }
 
     public Boolean hasEntry(String name) {
-        return null;
+        return phoneBook.entrySet().stream().anyMatch(entry -> IntStream.range(0, entry.getValue().size()).anyMatch(i -> entry.getValue().contains(name)));
```

Comment:
`Stream` uses an implementation of `Bui;der` pattern.
Thus, we express our method chains vertically by convention.
```
phoneBook
                .entrySet()
                .stream()
                .anyMatch(entry ->
                    IntStream.range(0, entry.getValue().size())
                            .anyMatch(i -> entry.getValue().contains(name));
```

However, I would prefer an implementation as follows:
```
        for (Map.Entry<String, List<String>> entry : phoneBook.entrySet()) {
            if (entry.getValue().contains(name)) {
                return true;
            }
        }
```